Revert "Add throttle for readMapOutput."

codexnull opened the pull request:
This reverts commit 491862602ed80fb8476e2ef529f75531e6ab4341.

External contribution appears to have broken the bulk loader. Didn't spend much time looking into why, just reverting it for now

Before the change:

```
REDUCE 01s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 02s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 03s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 04s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 05s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 06s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 07s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
...
```

After the change:

```
REDUCE 01s edge_count:574.1k edge_speed:574.1k/sec plist_count:516.8k plist_speed:516.8k/sec
REDUCE 02s edge_count:1.074M edge_speed:1.074M/sec plist_count:977.8k plist_speed:977.7k/sec
badger 2019/04/12 13:46:52 DEBUG: Storing value log head: {Fid:1 Len:45 Offset:681127}
REDUCE 03s edge_count:1.526M edge_speed:763.0k/sec plist_count:1.401M plist_speed:700.6k/sec
REDUCE 04s edge_count:2.026M edge_speed:675.3k/sec plist_count:1.876M plist_speed:625.1k/sec
...
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/3288)
<!-- Reviewable:end -->

mangalaman93 commented:
I don't think it is broken. I do notice that sometimes it starts with zero rate, but then I see positive numbers after a few seconds of wait. That also happens only sometimes, and I don't think that has anything to do with recent change. The recent change doesn't change the map phase to begin with. Do you see the rate to be zero continuously for some significant time (say more than 30 sec)?

This is the test that I ran -
```
aman@dgraph:~/gocode/src/github.com/dgraph-io/dgraph/dgraph$ git checkout origin/master
Note: checking out 'origin/master'.
HEAD is now at 5def6d7e docs: Update CHANGELOG for v1.0.14 and set default docs to v1.0.14.

aman@dgraph:~/gocode/src/github.com/dgraph-io/dgraph/dgraph$ make install
Old SHA256: 0e20d41549bb1cd9547cfcfaacfac8ba8d393c99cf00f9c4a1659924e01c2f7f
New SHA256: 0e20d41549bb1cd9547cfcfaacfac8ba8d393c99cf00f9c4a1659924e01c2f7f
aman@dgraph:~/gocode/src/github.com/dgraph-io/dgraph/dgraph$
```

Then I did a few runs -
```
aman@dgraph:~/gocode/src/github.com/dgraph-io/dgraph/systest/21million$ ./test-21million.sh 
INFO: entering directory .
INFO: removing old data (if any)
Stopping zero1 ... done
Removing zero1 ... done
Removing network dgraph_default
Removing volume dgraph_data
INFO: bringing up zero container
Creating network "dgraph_default" with the default driver
Creating volume "dgraph_data" with default driver
Creating zero1 ... done
INFO: waiting for zero to become leader
INFO: bulk loading data set
I0415 10:10:22.269876       8 init.go:88] 

Dgraph version   : v1.0.12-rc3-324-g5def6d7e
Commit SHA-1     : 5def6d7e
Commit timestamp : 2019-04-12 18:43:35 -0700
Branch           : HEAD
Go version       : go1.12.1

For Dgraph official documentation, visit https://docs.dgraph.io.
For discussions about Dgraph     , visit https://discuss.dgraph.io.
To say hi to the community       , visit https://dgraph.slack.com.

Licensed variously under the Apache Public License 2.0 and Dgraph Community License.
Copyright 2015-2018 Dgraph Labs, Inc.


{
	"DataFiles": "/dev/fd/62",
	"DataFormat": "rdf",
	"SchemaFile": "/dev/fd/63",
	"OutDir": "/data/alpha1/bulk",
	"ReplaceOutDir": false,
	"TmpDir": "tmp",
	"NumGoroutines": 16,
	"MapBufSize": 67108864,
	"ExpandEdges": true,
	"SkipMapPhase": false,
	"CleanupTmp": true,
	"NumShufflers": 1,
	"Version": false,
	"StoreXids": false,
	"ZeroAddr": "zero1:5080",
	"HttpAddr": "localhost:8080",
	"IgnoreErrors": false,
	"CustomTokenizers": "",
	"NewUids": false,
	"MapShards": 1,
	"ReduceShards": 1
}
Connecting to zero at zero1:5080
Processing file (1 out of 1): /dev/fd/62
MAP 02s nquad_count:213.0k err_count:0.000 nquad_speed:89.44k/sec edge_count:442.9k edge_speed:186.0k/sec
MAP 03s nquad_count:1.060M err_count:0.000 nquad_speed:313.5k/sec edge_count:2.201M edge_speed:650.9k/sec
MAP 04s nquad_count:1.870M err_count:0.000 nquad_speed:426.7k/sec edge_count:3.882M edge_speed:885.9k/sec
^C
```

```
aman@dgraph:~/gocode/src/github.com/dgraph-io/dgraph/systest/21million$ ./test-21million.sh 
INFO: entering directory .
INFO: removing old data (if any)
Stopping zero1 ... done
Removing zero1 ... done
Removing network dgraph_default
Removing volume dgraph_data
INFO: bringing up zero container
Creating network "dgraph_default" with the default driver
Creating volume "dgraph_data" with default driver
Creating zero1 ... done
INFO: waiting for zero to become leader
INFO: bulk loading data set
I0415 10:12:52.111598       8 init.go:88] 

Dgraph version   : v1.0.12-rc3-324-g5def6d7e
Commit SHA-1     : 5def6d7e
Commit timestamp : 2019-04-12 18:43:35 -0700
Branch           : HEAD
Go version       : go1.12.1

For Dgraph official documentation, visit https://docs.dgraph.io.
For discussions about Dgraph     , visit https://discuss.dgraph.io.
To say hi to the community       , visit https://dgraph.slack.com.

Licensed variously under the Apache Public License 2.0 and Dgraph Community License.
Copyright 2015-2018 Dgraph Labs, Inc.


{
	"DataFiles": "/dev/fd/62",
	"DataFormat": "rdf",
	"SchemaFile": "/dev/fd/63",
	"OutDir": "/data/alpha1/bulk",
	"ReplaceOutDir": false,
	"TmpDir": "tmp",
	"NumGoroutines": 16,
	"MapBufSize": 67108864,
	"ExpandEdges": true,
	"SkipMapPhase": false,
	"CleanupTmp": true,
	"NumShufflers": 1,
	"Version": false,
	"StoreXids": false,
	"ZeroAddr": "zero1:5080",
	"HttpAddr": "localhost:8080",
	"IgnoreErrors": false,
	"CustomTokenizers": "",
	"NewUids": false,
	"MapShards": 1,
	"ReduceShards": 1
}
Connecting to zero at zero1:5080
Processing file (1 out of 1): /dev/fd/62
MAP 02s nquad_count:137.7k err_count:0.000 nquad_speed:58.07k/sec edge_count:286.3k edge_speed:120.8k/sec
MAP 03s nquad_count:1.057M err_count:0.000 nquad_speed:313.5k/sec edge_count:2.194M edge_speed:650.9k/sec
MAP 04s nquad_count:1.891M err_count:0.000 nquad_speed:432.6k/sec edge_count:3.926M edge_speed:898.1k/sec
MAP 05s nquad_count:2.843M err_count:0.000 nquad_speed:529.3k/sec edge_count:5.903M edge_speed:1.099M/sec
MAP 06s nquad_count:3.730M err_count:0.000 nquad_speed:585.5k/sec edge_count:7.745M edge_speed:1.216M/sec
MAP 07s nquad_count:5.010M err_count:0.000 nquad_speed:678.7k/sec edge_count:10.40M edge_speed:1.409M/sec
MAP 08s nquad_count:5.697M err_count:0.000 nquad_speed:679.7k/sec edge_count:11.83M edge_speed:1.411M/sec
MAP 09s nquad_count:7.032M err_count:0.000 nquad_speed:749.5k/sec edge_count:14.60M edge_speed:1.556M/sec
MAP 10s nquad_count:7.774M err_count:0.000 nquad_speed:748.0k/sec edge_count:16.14M edge_speed:1.553M/sec
MAP 11s nquad_count:8.821M err_count:0.000 nquad_speed:774.3k/sec edge_count:18.32M edge_speed:1.608M/sec
MAP 12s nquad_count:9.569M err_count:0.000 nquad_speed:772.1k/sec edge_count:19.87M edge_speed:1.603M/sec
MAP 13s nquad_count:10.58M err_count:0.000 nquad_speed:790.1k/sec edge_count:21.98M edge_speed:1.641M/sec
^C
```

codexnull commented:
That's just the map phase. You need to wait for the REDUCE phase of the bulk load. Here's a sample run withthe dgraph binary built right after that commit (491862602ed80fb8476e2ef529f75531e6ab4341):

```
$ dgraph bulk -f 21million.rdf.gz -s 21million.schema
I0415 12:06:38.644470   24595 init.go:88] 

Dgraph version   : v1.0.12-rc3-310-g49186260
Commit SHA-1     : 49186260
Commit timestamp : 2019-04-12 02:36:11 +0530
Branch           : HEAD
Go version       : go1.12

For Dgraph official documentation, visit https://docs.dgraph.io.
For discussions about Dgraph     , visit https://discuss.dgraph.io.
To say hi to the community       , visit https://dgraph.slack.com.

Licensed variously under the Apache Public License 2.0 and Dgraph Community License.
Copyright 2015-2018 Dgraph Labs, Inc.


{
	"DataFiles": "21million.rdf.gz",
	"DataFormat": "",
	"SchemaFile": "21million.schema",
	"OutDir": "./out",
	"ReplaceOutDir": false,
	"TmpDir": "tmp",
	"NumGoroutines": 12,
	"MapBufSize": 67108864,
	"ExpandEdges": true,
	"SkipMapPhase": false,
	"CleanupTmp": true,
	"NumShufflers": 1,
	"Version": false,
	"StoreXids": false,
	"ZeroAddr": "localhost:5080",
	"HttpAddr": "localhost:8080",
	"IgnoreErrors": false,
	"CustomTokenizers": "",
	"NewUids": false,
	"MapShards": 1,
	"ReduceShards": 1
}

The bulk loader needs to open many files at once. This number depends on the size of the data set loaded, the map file output size, and the level of indexing. 100,000 is adequate for most data set sizes. See `man ulimit` for details of how to change the limit.
Current max open files limit: 1024

Connecting to zero at localhost:5080
Processing file (1 out of 1): 21million.rdf.gz
MAP 01s nquad_count:690.0k err_count:0.000 nquad_speed:689.7k/sec edge_count:1.432M edge_speed:1.431M/sec
MAP 02s nquad_count:1.814M err_count:0.000 nquad_speed:906.9k/sec edge_count:3.767M edge_speed:1.883M/sec
MAP 03s nquad_count:2.885M err_count:0.000 nquad_speed:961.3k/sec edge_count:5.989M edge_speed:1.996M/sec
MAP 04s nquad_count:4.173M err_count:0.000 nquad_speed:1.043M/sec edge_count:8.664M edge_speed:2.166M/sec
MAP 05s nquad_count:5.471M err_count:0.000 nquad_speed:1.094M/sec edge_count:11.36M edge_speed:2.271M/sec
MAP 06s nquad_count:6.779M err_count:0.000 nquad_speed:1.130M/sec edge_count:14.08M edge_speed:2.345M/sec
MAP 07s nquad_count:7.903M err_count:0.000 nquad_speed:1.129M/sec edge_count:16.41M edge_speed:2.344M/sec
MAP 08s nquad_count:8.912M err_count:0.000 nquad_speed:1.114M/sec edge_count:18.51M edge_speed:2.313M/sec
MAP 09s nquad_count:9.688M err_count:0.000 nquad_speed:1.076M/sec edge_count:20.12M edge_speed:2.235M/sec
MAP 10s nquad_count:10.18M err_count:0.000 nquad_speed:1.018M/sec edge_count:21.14M edge_speed:2.113M/sec
<SNIPPED>
MAP 01m00s nquad_count:21.24M err_count:0.000 nquad_speed:352.1k/sec edge_count:120.1M edge_speed:1.992M/sec
MAP 01m01s nquad_count:21.24M err_count:0.000 nquad_speed:346.4k/sec edge_count:120.1M edge_speed:1.959M/sec
MAP 01m02s nquad_count:21.24M err_count:0.000 nquad_speed:340.8k/sec edge_count:120.1M edge_speed:1.928M/sec
MAP 01m03s nquad_count:21.24M err_count:0.000 nquad_speed:335.4k/sec edge_count:120.1M edge_speed:1.897M/sec
MAP 01m04s nquad_count:21.24M err_count:0.000 nquad_speed:330.2k/sec edge_count:120.1M edge_speed:1.868M/sec
MAP 01m05s nquad_count:21.24M err_count:0.000 nquad_speed:325.2k/sec edge_count:120.1M edge_speed:1.839M/sec
MAP 01m06s nquad_count:21.24M err_count:0.000 nquad_speed:320.3k/sec edge_count:120.1M edge_speed:1.812M/sec
Shard tmp/shards/000 -> Reduce tmp/shards/shard_0/000
badger 2019/04/15 12:07:45 INFO: All 0 tables opened in 0s
REDUCE 01m07s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m08s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m09s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m10s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m11s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m12s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m13s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m14s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m15s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m16s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m17s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m18s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m19s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m20s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m21s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m22s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m23s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m24s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m25s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m26s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m27s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 01m28s [0.00%] edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
^C
```

After you've run the map phase once. You can skip it and jump right to the reduce phase. Here's a reduce phase run with the binary built at the previous commit:

```
$ dgraph bulk -f 21million.rdf.gz -s 21million.schema --skip_map_phase
I0415 12:18:38.565488   25390 init.go:88] 

Dgraph version   : v1.0.12-rc3-309-g3456be73
Commit SHA-1     : 3456be73
Commit timestamp : 2019-04-11 13:18:29 -0700
Branch           : HEAD
Go version       : go1.12

For Dgraph official documentation, visit https://docs.dgraph.io.
For discussions about Dgraph     , visit https://discuss.dgraph.io.
To say hi to the community       , visit https://dgraph.slack.com.

Licensed variously under the Apache Public License 2.0 and Dgraph Community License.
Copyright 2015-2018 Dgraph Labs, Inc.


{
	"DataFiles": "21million.rdf.gz",
	"DataFormat": "",
	"SchemaFile": "21million.schema",
	"OutDir": "./out",
	"ReplaceOutDir": false,
	"TmpDir": "tmp",
	"NumGoroutines": 12,
	"MapBufSize": 67108864,
	"ExpandEdges": true,
	"SkipMapPhase": true,
	"CleanupTmp": true,
	"NumShufflers": 1,
	"Version": false,
	"StoreXids": false,
	"ZeroAddr": "localhost:5080",
	"HttpAddr": "localhost:8080",
	"IgnoreErrors": false,
	"CustomTokenizers": "",
	"NewUids": false,
	"MapShards": 1,
	"ReduceShards": 1
}

The bulk loader needs to open many files at once. This number depends on the size of the data set loaded, the map file output size, and the level of indexing. 100,000 is adequate for most data set sizes. See `man ulimit` for details of how to change the limit.
Current max open files limit: 1024

Connecting to zero at localhost:5080
badger 2019/04/15 12:18:38 INFO: All 0 tables opened in 0s
REDUCE 01s edge_count:1.019M edge_speed:1.019M/sec plist_count:926.8k plist_speed:926.8k/sec
badger 2019/04/15 12:18:39 DEBUG: Storing value log head: {Fid:0 Len:42 Offset:57237428}
REDUCE 02s edge_count:1.858M edge_speed:1.858M/sec plist_count:1.716M plist_speed:1.715M/sec
badger 2019/04/15 12:18:41 DEBUG: Storing value log head: {Fid:1 Len:42 Offset:57041150}
REDUCE 03s edge_count:2.731M edge_speed:1.364M/sec plist_count:2.546M plist_speed:1.272M/sec
badger 2019/04/15 12:18:42 DEBUG: Storing value log head: {Fid:2 Len:42 Offset:56974243}
REDUCE 04s edge_count:3.658M edge_speed:1.218M/sec plist_count:3.427M plist_speed:1.141M/sec
badger 2019/04/15 12:18:43 DEBUG: Storing value log head: {Fid:3 Len:42 Offset:122510113}
REDUCE 05s edge_count:5.305M edge_speed:1.325M/sec plist_count:4.074M plist_speed:1.018M/sec
REDUCE 06s edge_count:6.830M edge_speed:1.365M/sec plist_count:4.943M plist_speed:988.0k/sec
badger 2019/04/15 12:18:44 DEBUG: Storing value log head: {Fid:4 Len:42 Offset:135087252}
badger 2019/04/15 12:18:45 INFO: Got compaction priority: {level:0 score:1 dropPrefix:[]}
badger 2019/04/15 12:18:45 INFO: Running for level: 0
REDUCE 07s edge_count:7.869M edge_speed:1.311M/sec plist_count:5.611M plist_speed:934.8k/sec
badger 2019/04/15 12:18:46 DEBUG: LOG Compact. Added 1925131 keys. Skipped 0 keys. Iteration took: 670.39201ms
REDUCE 08s edge_count:10.78M edge_speed:1.539M/sec plist_count:5.827M plist_speed:832.1k/sec
badger 2019/04/15 12:18:46 DEBUG: LOG Compact. Added 1928238 keys. Skipped 0 keys. Iteration took: 809.777813ms
badger 2019/04/15 12:18:47 DEBUG: Storing value log head: {Fid:5 Len:42 Offset:105293340}
badger 2019/04/15 12:18:47 DEBUG: LOG Compact. Added 1120331 keys. Skipped 0 keys. Iteration took: 403.100644ms
REDUCE 09s edge_count:12.68M edge_speed:1.584M/sec plist_count:5.975M plist_speed:746.6k/sec
badger 2019/04/15 12:18:47 DEBUG: Discard stats: map[]
badger 2019/04/15 12:18:47 INFO: LOG Compact 0->1, del 5 tables, add 3 tables, took 2.460695677s
badger 2019/04/15 12:18:47 INFO: Compaction for level: 0 DONE
REDUCE 10s edge_count:15.52M edge_speed:1.724M/sec plist_count:6.226M plist_speed:691.5k/sec
^C
```

And again after re-applying the commit:

```
$ dgraph bulk -f 21million.rdf.gz -s 21million.schema --skip_map_phase
I0415 12:20:58.255050   25782 init.go:88] 

Dgraph version   : v1.0.12-rc3-310-g49186260
Commit SHA-1     : 49186260
Commit timestamp : 2019-04-12 02:36:11 +0530
Branch           : HEAD
Go version       : go1.12

For Dgraph official documentation, visit https://docs.dgraph.io.
For discussions about Dgraph     , visit https://discuss.dgraph.io.
To say hi to the community       , visit https://dgraph.slack.com.

Licensed variously under the Apache Public License 2.0 and Dgraph Community License.
Copyright 2015-2018 Dgraph Labs, Inc.


{
	"DataFiles": "21million.rdf.gz",
	"DataFormat": "",
	"SchemaFile": "21million.schema",
	"OutDir": "./out",
	"ReplaceOutDir": false,
	"TmpDir": "tmp",
	"NumGoroutines": 12,
	"MapBufSize": 67108864,
	"ExpandEdges": true,
	"SkipMapPhase": true,
	"CleanupTmp": true,
	"NumShufflers": 1,
	"Version": false,
	"StoreXids": false,
	"ZeroAddr": "localhost:5080",
	"HttpAddr": "localhost:8080",
	"IgnoreErrors": false,
	"CustomTokenizers": "",
	"NewUids": false,
	"MapShards": 1,
	"ReduceShards": 1
}

The bulk loader needs to open many files at once. This number depends on the size of the data set loaded, the map file output size, and the level of indexing. 100,000 is adequate for most data set sizes. See `man ulimit` for details of how to change the limit.
Current max open files limit: 1024

Connecting to zero at localhost:5080
badger 2019/04/15 12:20:58 INFO: All 0 tables opened in 0s
REDUCE 01s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 02s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 03s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 04s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 05s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 06s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 07s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 08s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 09s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
REDUCE 10s edge_count:0.000 edge_speed:0.000/sec plist_count:0.000 plist_speed:0.000/sec
^C
```

You can also see the problem happen here when you expand the ./systest/21million/test-21million.sh output: https://teamcity.dgraph.io/viewLog.html?buildId=9616&buildTypeId=Dgraph_CiFullTestSuite&tab=buildLog&_focus=5816#_state=5045

mangalaman93 commented:
The patch that may fix the issue is below but @manishrjain mentioned that all the map files must be opened before shufflePostings starts. If we throttle the number of mappers and don't wait for them to finish (as the following patch proposes), we cannot guarantee that. On the other hand, if we do wait for them before starting shufflers, the mappers get blocked waiting for shufflers to read data from the channels. 

```
diff --git a/dgraph/cmd/bulk/shuffle.go b/dgraph/cmd/bulk/shuffle.go
index eb8d795a..8356c1b4 100644
--- a/dgraph/cmd/bulk/shuffle.go
+++ b/dgraph/cmd/bulk/shuffle.go
@@ -55,11 +55,12 @@ func (s *shuffler) run() {
 				go readMapOutput(mapFile, shuffleInputChs[i], readMapThr)
 			}
 
+			ci := &countIndexer{state: s.state, db: db}
+			s.shufflePostings(shuffleInputChs, ci)
+
 			// All map files must be opened before shufflePostings starts.
 			readMapThr.Wait()
 
-			ci := &countIndexer{state: s.state, db: db}
-			s.shufflePostings(shuffleInputChs, ci)
 			ci.wait()
 			thr.Done()
 		}(i, s.createBadger(i))
```
